codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDecoder.java

```diff
@@ -382,6 +396,28 @@ private void setDynamicTableSize(long dynamicTableSize) throws Http2Exception {
         hpackDynamicTable.setCapacity(dynamicTableSize);
     }
 
+    private Http2Headers.HeaderType validate(CharSequence name, Http2Headers.HeaderType previousHeaderType)
+        throws Http2Exception {
+        if (hasPseudoHeaderFormat(name)) {
```

Comment:
Netty typically allows users to opt out of header validation which may otherwise add overhead. For example in a "trusted" scenario it may be desirable to avoid the additional overhead (map lookup, string inspection, etc...). I would suggest we propagate the `validateHeaders` parameter in `DefaultHttp2HeadersDecoder` to this class and conditionally do the validation.

Reply:
- [X] Propage `validateHeaders`